github/Branch.py

```diff
@@ -298,23 +298,19 @@ def edit_protection(
                 post_parameters["required_pull_request_reviews"][
                     "bypass_pull_request_allowances"
                 ] = {}
-                if users_bypass_pull_request_allowances is github.GithubObject.NotSet:
-                    users_bypass_pull_request_allowances = []
-                if teams_bypass_pull_request_allowances is github.GithubObject.NotSet:
-                    teams_bypass_pull_request_allowances = []
-                if apps_bypass_pull_request_allowances is github.GithubObject.NotSet:
-                    apps_bypass_pull_request_allowances = []
+
+            if users_bypass_pull_request_allowances is not github.GithubObject.NotSet:
```

Comment:
Can we create a variable `bypass_pull_request_allowances` and split these in to 3 `if`?

from 
```python
if ... or ... or ... :
  if ...:
  if ...:
  if ...:
else:
   ...
```
to

```
bypass_pull_request_allowances = {}
if ...:
    ...
if ...:
    ... 
if ...:
    ...
if bypass_pull_request_allowances:
    ...
```

(I don't have write permission to this repo and I can't merge your PR, this is just my personal suggestion)

Reply:
I like that. @ThomasDevoogdt would you be happy to rework these ifs that construct elements `required_pull_request_reviews`, `dismissal_restrictions` and `bypass_pull_request_allowances`, as suggested?

Reply:
Reworked the commit that relates to `bypass_pull_request_allowances`, and added an extra commit that relates to `dismissal_restrictions`.